pkg/diff/diff_test.go

```diff
@@ -78,3 +85,185 @@ func (*testDiffSuite) TestGenerateSQLs(c *C) {
+	testCases := []struct {
+		createSourceTable string
+		createTargetTable string
+		dropSourceTable   string
+		dropTargetTable   string
+		structEqual       bool
+	}{
+		{
+			"CREATE TABLE `test`.`testa`(`id` int, `name` varchar(24))",
+			"CREATE TABLE `test`.`testb`(`id` int, `name` varchar(24), unique key (`id`))",
+			"DROP TABLE `test`.`testa`",
+			"DROP TABLE `test`.`testb`",
+			false,
+		}, {
+			"CREATE TABLE `test`.`testa`(`id` int, `name` varchar(24))",
+			"CREATE TABLE `test`.`testb`(`id` int, `name2` varchar(24))",
+			"DROP TABLE `test`.`testa`",
+			"DROP TABLE `test`.`testb`",
+			false,
+		}, {
+			"CREATE TABLE `test`.`testa`(`id` int, `name` varchar(24))",
+			"CREATE TABLE `test`.`testb`(`id` int)",
+			"DROP TABLE `test`.`testa`",
+			"DROP TABLE `test`.`testb`",
+			false,
+		}, {
+			"CREATE TABLE `test`.`testa`(`id` int, `name` varchar(24))",
+			"CREATE TABLE `test`.`testb`(`id` int, `name` varchar(24))",
+			"DROP TABLE `test`.`testa`",
+			"DROP TABLE `test`.`testb`",
+			true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		_, err := conn.Query(testCase.createSourceTable)
+		c.Assert(err, IsNil)
+		_, err = conn.Query(testCase.createTargetTable)
+		c.Assert(err, IsNil)
+
+		tableDiff := createTableDiff(conn)
+		structEqual, _, err := tableDiff.Equal(context.Background(), func(sql string) error {
+			fmt.Println(sql)
+			return nil
+		})
+		c.Assert(structEqual, Equals, testCase.structEqual)
+
+		_, err = conn.Query(testCase.dropSourceTable)
+		c.Assert(err, IsNil)
+		_, err = conn.Query(testCase.dropTargetTable)
+		c.Assert(err, IsNil)
+	}
+}
+
+func testDataEqual(dbConn *sql.DB, c *C) {
+	_, err := dbConn.Query(`
+		CREATE TABLE test.testa (
+			a date NOT NULL,
+			b datetime DEFAULT NULL,
+			c time DEFAULT NULL,
+			d varchar(10) COLLATE latin1_bin DEFAULT NULL,
+			e int(10) DEFAULT NULL,
+			h year(4) DEFAULT NULL,
+			PRIMARY KEY (a))
+	`)
+	c.Assert(err, IsNil)
+
+	_, err = dbConn.Query(`
+		CREATE TABLE test.testb (
+			a date NOT NULL,
+			b datetime DEFAULT NULL,
+			c time DEFAULT NULL,
+			d varchar(10) COLLATE latin1_bin DEFAULT NULL,
+			e int(10) DEFAULT NULL,
+			h year(4) DEFAULT NULL,
+			PRIMARY KEY (a))
+	`)
+	c.Assert(err, IsNil)
+
+	sql, err := ioutil.ReadFile("./test.sql")
+	c.Assert(err, IsNil)
+
+	_, err = dbConn.Exec(string(sql))
+	c.Assert(err, IsNil)
+
+	// load data to testb
+	_, err = dbConn.Exec("insert into test.testb (a, b, c, d, e, h) select a, b, c, d, e, h from test.testa")
+	c.Assert(err, IsNil)
+
+	fixSqls := make([]string, 0, 10)
+	writeSqls := func(sql string) error {
+		fixSqls = append(fixSqls, sql)
+		return nil
+	}
+
+	tableDiff := createTableDiff(dbConn)
+	structEqual, dataEqual, err := tableDiff.Equal(context.Background(), writeSqls)
+	c.Assert(err, IsNil)
+	c.Assert(structEqual, Equals, true)
+	c.Assert(dataEqual, Equals, true)
+
+	// update data and then compare data
+	_, err = dbConn.Exec("update test.testb set d = \"abc\" where a = \"2045-12-29\"")
```

Comment:
```suggestion
	_, err = dbConn.Exec("update test.testb set d = 'abc' where a = '2045-12-29'")
```

No need to escape :)